src/accounting/domain.ts

```
@@ -0,0 +1,83 @@
+    if (action.type === "증감") {
+      return acc.some((item) => item.type === action.target.type && item.id === action.target.id)
+        ? acc.map((item) =>
+            item.type === action.target.type && item.id === action.target.id
+              ? { ...item, amount: item.amount + action.amount }
+              : item
+          )
+        : [...acc, { type: action.target.type, id: action.target.id, amount: action.amount }];
+    }
+
+    // 예외 처리 강화
+    invariant(false, "지원하지 않는 트랜잭션 타입입니다.");
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)